[댓글 기능] 코맥 미션 제출합니다.

parker1609 opened the pull request:
이번 미션은 마감시한이 있어 최대한 이를 맞추기 위해 노력하였습니다. 감사합니다.

soojinroh committed:
[게시글 생성/조회/수정/삭제] 슬로스 미션 제출합니다. (#33)

* docs README.md 작성

* feat : 게시물의 정보를 담고 있는 bean 객체인 article class 구현

* docs: README.md 구현 기능 추가

* Update README.md

* Update README.md

* Update README.md

* feat 정적 파일 수정 시, 자동 재시작 및 LiveReload 설정

* feat 새로운 ArticleId를 생성하는 기능 구현

List<Article>에서 가장 큰 articleId보다 1만큼 큰 Long 타입 값을 반환

* fix: Article 클래스 수정

1. Article 클래스를 POJO로 수정
2. 불필요한 필드 삭제

* fix: articleId에 해당하는 Article 객체 조회 메서드 수정

* feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 조회 메서드 생성
2. 게시글 상세 조회 웹 UI 수정

* feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 상세 조회 메서드 구현
2. articleId에 해당하는 게시글을 가져와 Web UI에 구현

* feat: 웹 UI 수정

1. index.html, article.html, article-edit.html의 정적 파일 경로 수정
2. 전달받은 데이터를 타임리프를 이용하여 동적으로 출력하는 기능 적용

* feat : 게시물 수정 기능 구현

1. 수정하려는 게시글의 정보를 article-edit.html에 출력
2. 전달받은 Article 객체와 동일한 articleId를 가진 Article 객체를 List<Article>에서 바꾸는 메서드 구현
3. coverURL이 입력되지 않은 경우, 기본 이미지 URL 값을 할당하도록 수정

* refactor: 정적 파일의 경로를 수정

* refactor: 불필요한 파일 삭제

* feat : 환경설정 변경

* docs : README.md 수정

* feat : add lovely sloth image file

live like sloth, eat like sloth, sleep like sloth

* refactor : update REST 메서드 변경

1. post 방식의 메서드를 put으로 변경

* feat : 게시글 삭제 기능 구현

1. delete 메서드 방식을 이용해 게시글 삭제
2. article.html에 삭제 버튼 추가

* feat : 전체 게시글 목록 조회 기능 구현

1. get 메서드 방식을 이용해 전체 게시글 조회 메서드 구현
2. index.html에 전체 글 목록 출력

* refactor: css 오류 수정 및 게시글 상세 조회 페이지의 링크 경로 수정

* feat: ArticleRepository 테스트 추가

* refactor: ArticleController 클래스 테스트 리팩토링

1. 매직 넘버 상수화
2. response의 상태코드 확인 메서드 수정(is3xxRedirection() -> isFound())

* refactor : html 중복 제거

* refactor : convention 수정

1. 변수명 변경
2. 하드코딩한 값 상수화

* refactor : ArticleRepository의 DI를 필드 주입에서 생성자 주입으로 변경

* refactor gradle dependencies 주입 방식 변경

gradle의 dependencies 주입 방식을 compile에서 implementation으로 변경

* refactor url path 변경

CRUD 기능을 드러내던 url을 REST API naming convention에 따라 CRUD 기능을 드러내지 않는 것으로 수정

* feat matchId 메서드 추가

매개변수로 주어진 articleId와 해당 article의 articleId가 일치하는지 확인하는 matchId 메서드 구현

* refactor 변수 타입 변경

Article 클래스의 articleId의 변수 타입을 Long에서 long으로 변경

* refactor Article class의 setter 삭제

* feat ArticleDto 구현

* refactor ModelAttribute를 Article에서 ArticleDto로 수정

* refactor updateArticle 메서드 수정

list의 set 메서드를 이용해 article의 내용을 수정하던 것에서 Article의 update 메서드를 이용해 article을 수정하는 것으로 변경

* fix article id 변경

articleId를 빈 문자열에서 0으로 변경

* refactor test 코드 수정

soojinroh committed:
[게시판 회원 기능] 슬로스 미션 제출 합니다. (#100)

* revert

* [ADD] Add README

* [Fix] Code Formatting

* [Fix] Pass ArticleControllerTest, HelloWorldControllerTest

* [ADD] Move to writing page

* [ADD] Create article logic

* [ADD] Show article lists

* [ADD] Show selected article

* [ADD] Springboot Devtools (auto reload)

* [ADD] Remove duplication body code of index, article, article-edit HTML

- using Thymeleaf template layout

* [Fix] create_article Test

* [ADD] Remove duplicate head code of index, article, article-edit HTML

* [ADD] Modify article function

* [ADD] Remove article function

* [REF] Fix ArticleControllerTest

* [ADD] Redirect to article

* [REF] Add this keyword

* [REF] Extract magic number as constant variable

* [REF] Add matchId method in Article

* [ADD] ArticleRepository, ArticleController Test

* [REF] Improve ArticleRepository in ArticleController

- Modify field injection to constructor injection

* [Fix] add method in ArticleRepository

* [Fix] Remove this keyword

* [REF] Update 로직 수정, initial articleId 1로 수정

* [ADD] ArticleNotFoundException 추가

* [REF] ArticleControllerTests 중복제거

* Add : JPA 강의 적용

* Revert "Add : JPA 강의 적용"

This reverts commit f957b12deb23c177dd772a5430293452c62eaffb.

* Add : ArticleDto 생성

* Add : JPA-MySQL 설정 세팅

* Add : ArticleRepository 인터페이스 생성

* Refactor : Article class의 entity 속성 추가

* add : MySQL의 server time zone, UTF-8 encoding 설정 추가

* Refactor : 프론트 파일 교체

* Refactor : ArticleRepository를  CrudRepository를 상속하는 인터페이스로 변경

* Refactor : ArticleDto 삭제 및 ArticleController 수정

* Add : JPA Log 확인 설정 추가

* Refactor : ArticleControllerTests 수정

- 각 테스트마다 Article이 삽입 및 삭제되도록 수정

* Refactor : MainController 및 RequestMapping 분리

* Docs : README.md 수정

* Add : Login 페이지 이동 기능 구현

* Add : Signup 페이지 이동 기능 구현

* Add : User 클래스 생성

* Add : UserRequestDto 클래스 생성

* Add : UserRepository 인터페이스 생성

* Add : 회원가입 기능 구현

* Add : UserRequestDto Validation 패턴 추가

* Add : 회원가입 실패시 에러 메시지를 웹에 출력

* Add : UserControllerTests 회원가입 검증 테스트 추가

* Refactor : UserRequestDto 에러 메시지 상수화

* Add : signup.html 프론트 입력 검증 추가

* Add : signup.html 비밀번호 확인 기능 추가

* Add : 회원가입시 이메일 중복 확인 기능 추가

* Refactor : 이메일 중복 확인 기능 수정

* Add : 회원조회 기능 및 테스트 추가

* Add : 로그인, 로그아웃 기능 구현

* Refactor : 로그인 한 유저가 로그인시 메인화면으로 이동

* Refactor : Code Formatting

* Add, Refactor : 회원 탈퇴 기능 추가, 일부 메서드의 클래스 위치 변경

* Add : Logger 실습

* Refactor : HTML 중복 제거

* Add : user-list.html 중복 코드 제거

* Add : UserDetail 추가

* Add : mypage 이동 및 유저 정보 출력

* Add : mypage-edit 이동 및 유저 정보 출력

* Refactor : UserDetail 삭제

* Add : 회원정보 수정 기능 추가

* Add : 유저목록 탭 추가, delete버튼 UI 수정

* Add : 로그인여부에 따라 메인페이지 문구 출력

* Refactor : Controller 메서드 이전, 메서드 이름 수정

* Refactor : dto 패키지 분리

* Refactor : Controller에서 Service 분리

* Refactor : 사용하지않는 필드 삭제

* Refactor : 일부 테스트 메서드 이동 및 이름 수정

* Refactor : 패키지 및 파일 구조 변경

* Add : LoginControllerTests 구현

1. 로그인 성공
2. 로그인 실패(이메일 입력 오류)
3. 로그인 실패(패스워드 입력 오류)
4. 로그아웃 성공

* Refactor : package 구조 변경

* Add : ArticleServiceTests 구현

* Add : UserServiceTests 구현

* Add : LoginServiceTests 구현

* Refactor : annotation 수정

* Refactor : User Entity 수정

1. setter 제거
2. 기본 생성자의 접근제한자 private으로 변경
3. equals를 id만 비교하는 것으로 변경
4. column annotaion에서 name 제거
5. table annotation에서 name 제거

* Refactor : Article Entity 수정

1. setter 제거
2. 기본 생성자의 접근제한자 private으로 변경
3. equals를 id만 비교하는 것으로 변경

* Refactor : LoginRepository 삭제

* Refactor : 불필요한 Annotation 삭제

* Refactor : 메서드 변경

userRepository.findUserByEmail을 userRepository.existsByEmail로 변경

* Refactor : package 위치 변경

* Refactor : Controller에서 하던 Transaction관리를 service로 변경

* Refactor : ArticleUpdateDto 생성

* Refactor : Id type 변경

long에서 Long으로 변경

* Refactor : 사용하지 않는 method 삭제

* Refactor : ERROR MESSAGE 별도의 클래스로 분리

* Refactor : package 위치 변경

* Refactor : Convention 수정

1. @Autowired 작성하는 것으로 통일
2. ==를 equals()로 변경

* Add : Error message를 모아놓은 Messages 클래스 구현

* Refactor : html 파일 수정

* Refactor : 사용하지 않는 test 삭제

* Refactor : 주석 제거

* Test : test 코드 수정

indiveloper committed:
init myBlog step3

indiveloper committed:
[FEATURE] Apply Converter Pattern at article

indiveloper committed:
[CHORE] fix wrong words

indiveloper committed:
[REFACTOR] change from ArticleRequest to ArticleDto

indiveloper committed:
[REFACTOR] change from Converter to Assembler

indiveloper committed:
[REFACTOR] ArticleDto id 추가

indiveloper committed:
[REFACTOR] 자바 싱글턴 패턴에서 스프링 빈 등록

gch01410 committed:
[REFACTOR] user 클래스에 ModelMapper 적용

ModelMapper를 이용하여 user entity를 UserResponse, UserRequest로 변환
ModelMapper를 이용하여 UserResponse, UserRequest를 User entity로 변환

indiveloper committed:
[CHORE] server.port 설정 추가

8180으로 server.port 설정

indiveloper committed:
[FEATURE] Comment 클래스 구현

indiveloper committed:
[FEATURE] comment 생성 기능 구현

indiveloper committed:
[FIX] session 값 casting 변경

 session의 user attribute 값을 user에서 userResponse로 변경

indiveloper committed:
[REFACTOR] assembler 제거

ArticleService.java: assembler를 사용하여 articleDto에서 article로 변환하는 과정을 modelMapper를 사용하는 것으로 변경하였다.

indiveloper committed:
[FEATURE] Comment delete 기능 구현

작성자 확인 후 다르면 NotSameAuthorException 발생
Comment가 없으면 CommentNotFoundException 발생

indiveloper committed:
[FEATURE] comment update 기능 구현

indiveloper committed:
[FEATURE] 댓글 수정 및 삭제 버튼 유저 확인

Article.html: 해당 게시글에서 댓글 수정 및 삭제 버튼은 이를 작성한 회원만 보이도록 한다.

indiveloper committed:
[REFACTOR] InterceptorTest 로그인 기능 메소드로 분리

indiveloper committed:
[FEATURE] 로그인 여부에 따른 댓글 작성 Interceptor 경로 및 테스트 추가

indiveloper committed:
[TEST] 로그인하지 않은 경우에 대한 테스트 추가

indiveloper committed:
[REFACTOR] 프론트에 비밀번호 확인 기능 추가

백엔드에서 수행하던 비밀번호 확인 기능을 프론트엔드에서 수행하는 것으로 수정

indiveloper committed:
[FIX] UserController Test code 에러 수정

indiveloper committed:
[TEST] CommentControllerTests 구현

댓글 작성, 수정, 삭제, 조회 테스트 구현

indiveloper committed:
[REFACTOR] auto code formatting

indiveloper committed:
merge

indiveloper committed:
resolve conflict

phs1116 reviewed (changes requested):
안녕하세요 리뷰어 화투입니다!
전체적인 설계 및 구현 👍 
다만 몇가지 개선할 수 있는점과 테스트가 빠진 부분이 있어서 피드백 추가했습니다.
참고 부탁드리고 궁금하신 점 있으시면 DM주세요. :)

indiveloper committed:
[REFACTOR] Comment 삭제 및 수정 유저 확인 기능 서비스로 이동

indiveloper committed:
[REFACTOR] 패키지 이름 변경

service 패키지에서 application 패키지로 이름 변경

indiveloper committed:
[REFACTOR] Comment의 changeContents() 메서드 수정

Controller에서 수행하던 수정 및 삭제를 요청한 유저가 실제 해당 댓글을 작성한 유저인지 확인하는 로직을 Comment 클래스와 Service 클래스로 이동

indiveloper committed:
[REFACTOR] 사용하지 않는 클래스 및 메서드 삭제

indiveloper committed:
[FIX] Comment 수정 성공 테스트 에러 수정

indiveloper committed:
[FIX] Annotation 수정 및 주석 삭제

@Column 삭제 및 @JoinColumn에 (nullable=false) 추가

indiveloper committed:
[REFACTOR] StringUtils.isBlank() 적용

indiveloper committed:
[REFACTOR] Article entity에 User author 필드 추가 및 그에 따른 테스트 수정

indiveloper committed:
[REFACTOR] findUserByEmail을 findById로 수정

indiveloper committed:
[FEATURE] CASCADE annotation 추가

Article과 Comment에 CASCASDE annotation 추가

indiveloper committed:
[REFACTOR] 사용하지 않는 생성자 삭제

indiveloper committed:
[REFACTOR] return하는 List를 Collections.unmodifiableList로 변경

indiveloper committed:
[REFACTOR] Comment class의 isSameAuthor 메서드 기능 수정

indiveloper committed:
[REFACTOR] RequestMapping으로 중복 url 제거

parker1609 committed:
[REFACTOR] merge 후 변경되지 않은 사항 수정

parker1609 committed:
[REFACTOR] 길어지는 코드 줄바꿈

parker1609 committed:
[REFACTOR] repository 패키지 분리

parker1609 committed:
[REFACTOR] 데이터베이스 설정 변경

parker1609 committed:
[REFACTOR] Article Controller에서 article 수정 페이지 접근 확인 기능 추가

parker1609 committed:
[REFACTOR] ExceptionHandler 추가

parker1609 committed:
[REFACTOR] 댓글 수정 및 삭제 시 유저 확인 로직 변경

parker1609 committed:
[TEST] CommentService 테스트 구현

parker1609 committed:
[REFACTOR] 오타 수정

parker1609 committed:
[REFACTOR] interceptor 수정

parker1609 committed:
[TEST] ArticleService 테스트 추가

parker1609 committed:
[TEST] UserService 테스트 추가

parker1609 committed:
[TEST] 댓글 오류 테스트 추가

parker1609 committed:
[CHORE] ApplicationRunner 클래스 추가

parker1609 committed:
[REFACTOR] 공통 부분 추상 클래스로 분리

parker1609 committed:
[REFACTOR] 사용하지 않는 import문 및 들여쓰기 맞춤

phs1116 reviewed (approved):
안녕하세요! 리뷰어 화투입니다.
코드가 상당히 많이 변경되었네요. 👍 
전체적으로 잘 구현 하신 것 같습니다.
이번 단계 머지하도록 할게요.
다만 몇가지 피드백 추가했는데 다음 단계 진행하시면서 참고 부탁드립니다. :)
